File: 2023_algo_ds/practise/containsDuplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
function containsDuplicate(nums) {
    const lookup = new Set();
    for(const n of nums){
        if(lookup.has(n)){
            return true;
        }else{
            lookup.add(n);
        }
    }

    return false;
}
// function containsDuplicate(nums) {
//    return nums.length !== new Set([...nums]).size
// }
// function containsDuplicate(nums) {
//     return !!Object.values(nums.reduce((acc, val) => {
//         if(acc?.[val]){
//             acc[val] += 1;
//         }else {
//             acc[val] = 1;
//         }
//         return acc; 
//     }, {}))
//     .find(v=>v>1);
// }






// var containsDuplicate = function(nums) {
//     let lookup = new Map();
//     let containsDup = false;
//     let map = nums.forEach(n=> {
//         lookup.set(n, (lookup.get(n) ?? 0 ) + 1);
//         if(lookup.get(n) > 1){
//             containsDup = true;
//         }
//     });
//     return containsDup;
    
// };

File: 2024_dsa/practise/array/contains_duplicate.py
from typing import List


class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        lookup = {}
        for n in nums:
            lookup[n] = lookup.get(n, 0) + 1
            print(lookup)
            print(lookup.get(n, 0))
            if lookup.get(n, 0) > 1:
                return True
        return False


dup = Solution().containsDuplicate(
    [
        1,
        2,
        3,
        4,
        5,
        6,
        6,
    ]
)

print(dup)
